tests/app/views/test_search_queries.py

```diff
@@ -0,0 +1,183 @@
+from app.main.services import search_service
+from flask import json
+from nose.tools import assert_equal, ok_
+
+from app import create_app
+from ..helpers import setup_authorization, teardown_authorization
+from ..helpers import default_service
+
+
+def test_single_filter_queries():
+    yield check_query, '', 120, {}
+    yield check_query, 'filter_lot=SaaS', 30, {'lot': matches("SaaS")}
+    yield (check_query, 'filter_serviceTypes=Implementation',
+           48, {'serviceTypes': contains('Implementation')})
+    yield count_for_query, 'filter_minimumContractPeriod=Hour', 40
```

Comment:
In some places `count_for_query` is used and in others `check_query` with an empty `match_fields`.